### ISSUE TITLE
Feature request: Search on multiple terms

### ISSUE THREAD

**Issue by garryknight** [friendica/friendica#7048]
Recently I wanted to search for all posts by one particular user - my doppelganger on Diaspora* - which had a particular hashtag attached to those posts. With the resulting list, I wanted to add each post to a Saved Folder, which is something I can't currently do on Diaspora*.

I tried entering an @user@pod name and a #hashtag into the Search field but, unsurprisingly, got a "No matches" message.

Is it possible that this feature could be added, please?

**Comment by nupplaphil** [comment]
Since I do want to refactor the search logic, I'll take this feature request and try to include it as well.

(Will be part of moving mod/search and mod/ match)

**Comment by annando** [comment]
We now do have the full text search with a lot of nice options to include and exclude search terms. So I think that can be declared as solved.

**Comment by MrPetovan** [comment]
Not so fast, the results are ordered by time instead of relevance which makes the multiple terms search irrelevant.

**Comment by annando** [comment]
Well, including and excluding multiple search terms works very well.

**Comment by MrPetovan** [comment]
Of course, it works technically, but if you can't use the results because out of the the two terms you provided, one of them appears more recently alone than the two together, I don't consider it solved.

**Comment by annando** [comment]
We are using the "Boolean Full-Text Searches" which means: "MySQL can perform boolean full-text searches using the IN BOOLEAN MODE modifier. With this modifier, certain characters have special meaning at the beginning or end of words in the search string. In the following query, the + and - operators indicate that a word must be present or absent, respectively, for a match to occur. "

See here: https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html

The boolean mode has got this characteristic according to that page: "They do not automatically sort rows in order of decreasing relevance."

The boolean mode is used here because of the possibility of modifiers like `+` or `-`, use of parentheses and some other cool stuff.

**Comment by MrPetovan** [comment]
Yup, I know about it, and yet I was surprised that the first results of my search didn't contain both terms I typed. When I search for "term1 term2", I expect results including both terms to be shown first before "term1" alone or "term2" alone, without having to use a fancy syntax I know about but that we don't explain anywhere and shouldn't be necessary in this case.

**Comment by annando** [comment]
I'm unsure how to better document this behaviour. But nonetheless: Because of the way we are performing this query (as a query inside a query where the result is used for the main query) we wouldn't be able to profit from the sorting of the result set. I also don't see a possibility to change this without having a massive speed impact on this.

**Comment by MrPetovan** [comment]
Whether this can be improved with documentation only or with a technical change, however complex and performance-impacting it is (or both!) isn't the purpose of my original comment, just that I don't feel like this issue has been completely addressed.
